crates/file/src/blob.rs

```diff
@@ -0,0 +1,309 @@
+    {
+        Blob::new_with_options(content, None)
+    }
+
+    /// Like `new`, but allows specifying the MIME type (also known as *content type* or *media
+    /// type*) of the `Blob`.
+    pub fn new_with_options<T>(content: T, mime_type: Option<&str>) -> Blob
+    where
+        T: BlobContents,
+    {
+        let mut properties = web_sys::BlobPropertyBag::new();
+        if let Some(mime_type) = mime_type {
+            properties.type_(mime_type);
+        }
+
+        let parts = js_sys::Array::of1(&content.into_jsvalue());
+        let inner = web_sys::Blob::new_with_u8_array_sequence_and_options(&parts, &properties);
+
+        Blob::from(inner.unwrap_throw())
+    }
+
+    pub fn slice(&self, start: u64, end: u64) -> Self {
+        let start = safe_u64_to_f64(start);
+        let end = safe_u64_to_f64(end);
+
+        let b: &web_sys::Blob = self.as_ref();
+        Blob::from(b.slice_with_f64_and_f64(start, end).unwrap_throw())
+    }
+
+    /// The number of bytes in the Blob/File.
+    pub fn size(&self) -> u64 {
+        safe_f64_to_u64(self.inner.size())
+    }
+
+    /// The statically typed MIME type (also known as *content type* or *media type*) of the `File`
+    /// or `Blob`.
+    #[cfg(feature = "mime")]
+    pub fn mime_type(&self) -> Result<mime::Mime, mime::FromStrError> {
+        self.raw_mime_type().parse()
+    }
+
+    /// The raw MIME type (also known as *content type* or *media type*) of the `File` or
+    /// `Blob`.
+    pub fn raw_mime_type(&self) -> String {
+        self.inner.type_()
+    }
+}
+
+impl From<web_sys::Blob> for Blob {
+    fn from(blob: web_sys::Blob) -> Self {
+        Blob { inner: blob }
+    }
+}
+
+impl From<web_sys::File> for Blob {
+    fn from(file: web_sys::File) -> Self {
+        Blob { inner: file.into() }
+    }
+}
+
+impl From<Blob> for web_sys::Blob {
+    fn from(blob: Blob) -> Self {
+        blob.inner
+    }
+}
+
+impl From<Blob> for JsValue {
+    fn from(blob: Blob) -> Self {
+        blob.inner.into()
+    }
+}
+
+impl AsRef<web_sys::Blob> for Blob {
+    fn as_ref(&self) -> &web_sys::Blob {
+        self.inner.as_ref()
+    }
+}
+
+impl AsRef<JsValue> for Blob {
+    fn as_ref(&self) -> &JsValue {
+        self.inner.as_ref()
+    }
+}
+
+/// A [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File).
+#[derive(Debug, Clone)]
+pub struct File {
+    // the trick here is that we know the contents of `inner` are a file, even though that type
+    // information is not stored. It is the same trick as is used in `web_sys`.
+    inner: Blob,
+}
+
+impl File {
+    /// Create a new `File` with the given name and contents.
+    ///
+    /// `contents` can be from one of `&str`, `&[u8]`, or others.
+    pub fn new<T>(name: &str, contents: T) -> File
+    where
+        T: BlobContents,
+    {
+        Self::new_(name, contents.into_jsvalue(), None, None)
+    }
+
+    /// Like `File::new`, but allows customizing the last modified time and the MIME type (also
```

Comment:
Docs order should match the argument order, mentioning MIME type first and then last modified time.